{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "b7b6a6c7-5316-4ff4-abbb-ee335b8e54b5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to work/vcf/h3a_wgs.log.\n",
      "Options in effect:\n",
      "  --bfile work/binary/h3a_wgs\n",
      "  --out work/vcf/h3a_wgs\n",
      "  --recode vcf-iid\n",
      "\n",
      "192825 MB RAM detected; reserving 96412 MB for main workspace.\n",
      "1252340 variants loaded from .bim file.\n",
      "386 people (0 males, 0 females, 386 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to work/vcf/h3a_wgs.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 386 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is exactly 1.\n",
      "1252340 variants and 386 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--recode vcf-iid to work/vcf/h3a_wgs.vcf ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "CPU times: user 263 ms, sys: 85 ms, total: 348 ms\n",
      "Wall time: 13.1 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "!plink --bfile work/binary/h3a_wgs --recode vcf-iid --out work/vcf/h3a_wgs "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "b75684c6-2fd1-4b2c-8a89-b5a33d1eff64",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf work/vcf/h3a_wgs.vcf\n",
      "\t--weir-fst-pop countries/DRC.txt\n",
      "\t--weir-fst-pop countries/Uganda.txt\n",
      "\t--keep countries/DRC.txt\n",
      "\t--keep countries/Uganda.txt\n",
      "\t--out output/fst/drc_vs_uganda\n",
      "\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 63 out of 386 Individuals\n",
      "Outputting Weir and Cockerham Fst estimates.\n",
      "Weir and Cockerham mean Fst estimate: 0.0071302\n",
      "Weir and Cockerham weighted Fst estimate: 0.0085369\n",
      "After filtering, kept 1252340 out of a possible 1252340 Sites\n",
      "Run Time = 134.00 seconds\n",
      "CPU times: user 2.13 s, sys: 366 ms, total: 2.5 s\n",
      "Wall time: 2min 14s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "!vcftools --vcf work/vcf/h3a_wgs.vcf --weir-fst-pop countries/DRC.txt --weir-fst-pop countries/Uganda.txt --out output/fst/drc_vs_uganda"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "b0722fa7-56ef-459e-a996-4b9883f7c552",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf work/vcf/h3a_wgs.vcf\n",
      "\t--weir-fst-pop countries/Uganda.txt\n",
      "\t--weir-fst-pop countries/Botswana.txt\n",
      "\t--keep countries/Uganda.txt\n",
      "\t--keep countries/Botswana.txt\n",
      "\t--out output/fst/uganda_vs_botswana\n",
      "\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 99 out of 386 Individuals\n",
      "Outputting Weir and Cockerham Fst estimates.\n",
      "Weir and Cockerham mean Fst estimate: 0.013809\n",
      "Weir and Cockerham weighted Fst estimate: 0.015672\n",
      "After filtering, kept 1252340 out of a possible 1252340 Sites\n",
      "Run Time = 136.00 seconds\n",
      "CPU times: user 2.1 s, sys: 470 ms, total: 2.57 s\n",
      "Wall time: 2min 15s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "!vcftools --vcf work/vcf/h3a_wgs.vcf --weir-fst-pop countries/Uganda.txt --weir-fst-pop countries/Botswana.txt --out output/fst/uganda_vs_botswana"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "2b1d7a01-dea5-462c-85ac-d9523f966de3",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "5be7c02e-6721-4bd3-861b-4f5d183a5847",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SampleID</th>\n",
       "      <th>cluster</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>H3A-7C44F2</td>\n",
       "      <td>Western</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>H3A-B4DD3C</td>\n",
       "      <td>Western</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>H3A-31F23C</td>\n",
       "      <td>Eastern</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>H3A-D3A6FE</td>\n",
       "      <td>Eastern</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>H3A-0BF47F</td>\n",
       "      <td>Western</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     SampleID  cluster\n",
       "0  H3A-7C44F2  Western\n",
       "1  H3A-B4DD3C  Western\n",
       "2  H3A-31F23C  Eastern\n",
       "3  H3A-D3A6FE  Eastern\n",
       "4  H3A-0BF47F  Western"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "regions = pd.read_csv('/data/vcf/h3awgs/clusters.tsv', sep='\\t')\n",
    "regions.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "d67e899a-952f-4fd0-a30d-fca4f378a9ab",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Country</th>\n",
       "      <th>EthnolinguisticGroup</th>\n",
       "      <th>Gender</th>\n",
       "      <th>PI</th>\n",
       "      <th>Project</th>\n",
       "      <th>sourcefile</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SampleID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>H3A-7C44F2</th>\n",
       "      <td>Guinea</td>\n",
       "      <td>Guinea</td>\n",
       "      <td>Female</td>\n",
       "      <td>Fowler</td>\n",
       "      <td>Trypanogen</td>\n",
       "      <td>Trypanogen.tsv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H3A-B4DD3C</th>\n",
       "      <td>Cote D'Ivoire</td>\n",
       "      <td>Cote D'Ivoire</td>\n",
       "      <td>Male</td>\n",
       "      <td>Fowler</td>\n",
       "      <td>Trypanogen</td>\n",
       "      <td>Trypanogen.tsv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H3A-31F23C</th>\n",
       "      <td>Uganda</td>\n",
       "      <td>Uganda</td>\n",
       "      <td>Female</td>\n",
       "      <td>Fowler</td>\n",
       "      <td>Trypanogen</td>\n",
       "      <td>Trypanogen.tsv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H3A-D3A6FE</th>\n",
       "      <td>Uganda</td>\n",
       "      <td>Uganda</td>\n",
       "      <td>Female</td>\n",
       "      <td>Fowler</td>\n",
       "      <td>Trypanogen</td>\n",
       "      <td>Trypanogen.tsv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H3A-0BF47F</th>\n",
       "      <td>Guinea</td>\n",
       "      <td>Guinea</td>\n",
       "      <td>Male</td>\n",
       "      <td>Fowler</td>\n",
       "      <td>Trypanogen</td>\n",
       "      <td>Trypanogen.tsv</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Country EthnolinguisticGroup  Gender      PI     Project  \\\n",
       "SampleID                                                                     \n",
       "H3A-7C44F2         Guinea               Guinea  Female  Fowler  Trypanogen   \n",
       "H3A-B4DD3C  Cote D'Ivoire        Cote D'Ivoire    Male  Fowler  Trypanogen   \n",
       "H3A-31F23C         Uganda               Uganda  Female  Fowler  Trypanogen   \n",
       "H3A-D3A6FE         Uganda               Uganda  Female  Fowler  Trypanogen   \n",
       "H3A-0BF47F         Guinea               Guinea    Male  Fowler  Trypanogen   \n",
       "\n",
       "                sourcefile  \n",
       "SampleID                    \n",
       "H3A-7C44F2  Trypanogen.tsv  \n",
       "H3A-B4DD3C  Trypanogen.tsv  \n",
       "H3A-31F23C  Trypanogen.tsv  \n",
       "H3A-D3A6FE  Trypanogen.tsv  \n",
       "H3A-0BF47F  Trypanogen.tsv  "
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta  = pd.read_csv('/data/vcf/h3awgs/samples.meta.tsv', sep='\\t', index_col='SampleID')\n",
    "meta.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "556fbad3-d151-424a-b80c-5f02a07c92a7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SampleID</th>\n",
       "      <th>cluster</th>\n",
       "      <th>Country</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>H3A-7C44F2</td>\n",
       "      <td>Western</td>\n",
       "      <td>Guinea</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>H3A-B4DD3C</td>\n",
       "      <td>Western</td>\n",
       "      <td>Cote D'Ivoire</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>H3A-31F23C</td>\n",
       "      <td>Eastern</td>\n",
       "      <td>Uganda</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>H3A-D3A6FE</td>\n",
       "      <td>Eastern</td>\n",
       "      <td>Uganda</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>H3A-0BF47F</td>\n",
       "      <td>Western</td>\n",
       "      <td>Guinea</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     SampleID  cluster        Country\n",
       "0  H3A-7C44F2  Western         Guinea\n",
       "1  H3A-B4DD3C  Western  Cote D'Ivoire\n",
       "2  H3A-31F23C  Eastern         Uganda\n",
       "3  H3A-D3A6FE  Eastern         Uganda\n",
       "4  H3A-0BF47F  Western         Guinea"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta_dict = meta.iloc[:,[0]].to_dict()\n",
    "countries = meta_dict.get('Country')\n",
    "regions[\"Country\"] =  regions['SampleID'].map(lambda x: countries.get(x))\n",
    "regions.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "id": "0710a5a9-a1cb-4e85-a4ff-1effa308961f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Western cluster ['Guinea' \"Cote D'Ivoire\" 'Benin' 'Nigeria' 'Cameroon' 'Burkina Faso'\n",
      " 'Ghana']\n",
      "Eastern cluster ['Uganda' 'Zambia']\n",
      "Central cluster ['DRC']\n",
      "Southern cluster ['Botswana']\n"
     ]
    }
   ],
   "source": [
    "region_clusters = {}\n",
    "for region in regions.cluster.unique():\n",
    "    df = regions[regions['cluster'] == region]\n",
    "    region_clusters[region] = df.Country.unique()\n",
    "    print(f'{region} cluster',df.Country.unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "57402ba8-1f6c-4d4a-8182-2829fd8fad02",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf work/vcf/h3a_wgs.vcf\n",
      "\t--weir-fst-pop countries/Guinea.txt\n",
      "\t--weir-fst-pop countries/Cote D'Ivoire.txt\n",
      "\t--weir-fst-pop countries/Benin.txt\n",
      "\t--weir-fst-pop countries/Nigeria.txt\n",
      "\t--weir-fst-pop countries/Cameroon.txt\n",
      "\t--weir-fst-pop countries/Burkina Faso.txt\n",
      "\t--weir-fst-pop countries/Ghana.txt\n",
      "\t--keep countries/Guinea.txt\n",
      "\t--keep countries/Cote D'Ivoire.txt\n",
      "\t--keep countries/Benin.txt\n",
      "\t--keep countries/Nigeria.txt\n",
      "\t--keep countries/Cameroon.txt\n",
      "\t--keep countries/Burkina Faso.txt\n",
      "\t--keep countries/Ghana.txt\n",
      "\t--out output/fst/Western\n",
      "\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 246 out of 386 Individuals\n",
      "Outputting Weir and Cockerham Fst estimates.\n",
      "Weir and Cockerham mean Fst estimate: 0.0050698\n",
      "Weir and Cockerham weighted Fst estimate: 0.0054486\n",
      "After filtering, kept 1252340 out of a possible 1252340 Sites\n",
      "Run Time = 146.00 seconds\n",
      "CPU times: user 2.25 s, sys: 468 ms, total: 2.72 s\n",
      "Wall time: 2min 26s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "!vcftools --vcf work/vcf/h3a_wgs.vcf --weir-fst-pop countries/Guinea.txt --weir-fst-pop countries/Cote\\ D\\'Ivoire.txt --weir-fst-pop countries/Benin.txt --weir-fst-pop countries/Nigeria.txt --weir-fst-pop countries/Cameroon.txt --weir-fst-pop countries/Burkina\\ Faso.txt --weir-fst-pop countries/Ghana.txt --out output/fst/Western"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "5b2c5e8e-c995-4d82-a4ed-b3e4608d8490",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf work/vcf/h3a_wgs.vcf\n",
      "\t--weir-fst-pop countries/Zambia.txt\n",
      "\t--weir-fst-pop countries/Uganda.txt\n",
      "\t--keep countries/Zambia.txt\n",
      "\t--keep countries/Uganda.txt\n",
      "\t--out output/fst/Eastern\n",
      "\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 80 out of 386 Individuals\n",
      "Outputting Weir and Cockerham Fst estimates.\n",
      "Weir and Cockerham mean Fst estimate: 0.011402\n",
      "Weir and Cockerham weighted Fst estimate: 0.012212\n",
      "After filtering, kept 1252340 out of a possible 1252340 Sites\n",
      "Run Time = 135.00 seconds\n",
      "CPU times: user 2.1 s, sys: 431 ms, total: 2.53 s\n",
      "Wall time: 2min 14s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "!vcftools --vcf work/vcf/h3a_wgs.vcf --weir-fst-pop countries/Zambia.txt --weir-fst-pop countries/Uganda.txt --out output/fst/Eastern"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "dbe22471-b15c-4647-a713-40e38369250b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf work/vcf/h3a_wgs.vcf\n",
      "\t--weir-fst-pop countries/DRC.txt\n",
      "\t--weir-fst-pop countries/DRC.txt\n",
      "\t--keep countries/DRC.txt\n",
      "\t--keep countries/DRC.txt\n",
      "\t--out output/fst/Central\n",
      "\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 12 out of 386 Individuals\n",
      "Outputting Weir and Cockerham Fst estimates.\n",
      "Weir and Cockerham mean Fst estimate: -0.046509\n",
      "Weir and Cockerham weighted Fst estimate: -0.047356\n",
      "After filtering, kept 1252340 out of a possible 1252340 Sites\n",
      "Run Time = 132.00 seconds\n",
      "CPU times: user 2.06 s, sys: 469 ms, total: 2.53 s\n",
      "Wall time: 2min 12s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "!vcftools --vcf work/vcf/h3a_wgs.vcf --weir-fst-pop countries/DRC.txt --weir-fst-pop countries/DRC.txt --out output/fst/Central"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "34b689eb-ed74-4a3f-9513-921960a9cd70",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf work/vcf/h3a_wgs.vcf\n",
      "\t--weir-fst-pop countries/D1.txt\n",
      "\t--weir-fst-pop countries/D2.txt\n",
      "\t--keep countries/D1.txt\n",
      "\t--keep countries/D2.txt\n",
      "\t--out output/fst/Central2\n",
      "\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 12 out of 386 Individuals\n",
      "Outputting Weir and Cockerham Fst estimates.\n",
      "Weir and Cockerham mean Fst estimate: -0.046509\n",
      "Weir and Cockerham weighted Fst estimate: -0.047356\n",
      "After filtering, kept 1252340 out of a possible 1252340 Sites\n",
      "Run Time = 132.00 seconds\n",
      "CPU times: user 2.03 s, sys: 401 ms, total: 2.43 s\n",
      "Wall time: 2min 12s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "!head -n 24 countries/DRC.txt > countries/D1.txt\n",
    "!tail -n 24 countries/DRC.txt > countries/D2.txt\n",
    "\n",
    "!vcftools --vcf work/vcf/h3a_wgs.vcf --weir-fst-pop countries/D1.txt --weir-fst-pop countries/D2.txt --out output/fst/Central2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "id": "beae82fd-2b7b-4cc8-880f-4870cfa02f98",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf work/vcf/h3a_wgs.vcf\n",
      "\t--weir-fst-pop countries/Botswana.txt\n",
      "\t--weir-fst-pop countries/Botswana.txt\n",
      "\t--keep countries/Botswana.txt\n",
      "\t--keep countries/Botswana.txt\n",
      "\t--out output/fst/Southern\n",
      "\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 48 out of 386 Individuals\n",
      "Outputting Weir and Cockerham Fst estimates.\n",
      "Weir and Cockerham mean Fst estimate: -0.010364\n",
      "Weir and Cockerham weighted Fst estimate: -0.010338\n",
      "After filtering, kept 1252340 out of a possible 1252340 Sites\n",
      "Run Time = 135.00 seconds\n",
      "CPU times: user 2.01 s, sys: 404 ms, total: 2.42 s\n",
      "Wall time: 2min 15s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "!vcftools --vcf work/vcf/h3a_wgs.vcf --weir-fst-pop countries/Botswana.txt --weir-fst-pop countries/Botswana.txt --out output/fst/Southern"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "f8d102d9-a405-4126-a205-7851bbe01f9c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf work/vcf/h3a_wgs.vcf\n",
      "\t--weir-fst-pop countries/B1.txt\n",
      "\t--weir-fst-pop countries/B2.txt\n",
      "\t--keep countries/B1.txt\n",
      "\t--keep countries/B2.txt\n",
      "\t--out output/fst/Southern2\n",
      "\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 48 out of 386 Individuals\n",
      "Outputting Weir and Cockerham Fst estimates.\n",
      "Weir and Cockerham mean Fst estimate: -0.00034047\n",
      "Weir and Cockerham weighted Fst estimate: 3.2011e-05\n",
      "After filtering, kept 1252340 out of a possible 1252340 Sites\n",
      "Run Time = 133.00 seconds\n",
      "CPU times: user 2.07 s, sys: 370 ms, total: 2.44 s\n",
      "Wall time: 2min 13s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "!head -n 24 countries/Botswana.txt > countries/B1.txt\n",
    "!tail -n 24 countries/Botswana.txt > countries/B2.txt\n",
    "\n",
    "!vcftools --vcf work/vcf/h3a_wgs.vcf --weir-fst-pop countries/B1.txt --weir-fst-pop countries/B2.txt --out output/fst/Southern2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "192c2ddc-8e38-4a26-bb8a-2400d1f6f883",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SampleID</th>\n",
       "      <th>Cluster</th>\n",
       "      <th>Country</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>H3A-7C44F2</td>\n",
       "      <td>Delta</td>\n",
       "      <td>Guinea</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>H3A-B4DD3C</td>\n",
       "      <td>Delta</td>\n",
       "      <td>Cote D'Ivoire</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>H3A-31F23C</td>\n",
       "      <td>Beta</td>\n",
       "      <td>Uganda</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>H3A-D3A6FE</td>\n",
       "      <td>Beta</td>\n",
       "      <td>Uganda</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>H3A-0BF47F</td>\n",
       "      <td>Delta</td>\n",
       "      <td>Guinea</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     SampleID Cluster        Country\n",
       "0  H3A-7C44F2   Delta         Guinea\n",
       "1  H3A-B4DD3C   Delta  Cote D'Ivoire\n",
       "2  H3A-31F23C    Beta         Uganda\n",
       "3  H3A-D3A6FE    Beta         Uganda\n",
       "4  H3A-0BF47F   Delta         Guinea"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clusters = pd.read_csv('output/cluster/cluster.tsv', sep='\\t')\n",
    "clusters.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "4f3732ea-b770-4e3e-9314-60c93528c99d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Delta cluster ['Guinea' \"Cote D'Ivoire\" 'DRC']\n",
      "Beta cluster ['Uganda']\n",
      "Gamma cluster ['Benin' 'Nigeria' 'Burkina Faso' 'Ghana' 'Cameroon']\n",
      "Epsilon cluster ['Botswana']\n",
      "Alpha cluster ['Botswana' 'Cameroon' 'Zambia']\n"
     ]
    }
   ],
   "source": [
    "computational_clusters = {}\n",
    "for cluster in clusters.Cluster.unique():\n",
    "    df = clusters[clusters['Cluster'] == cluster]\n",
    "    computational_clusters[cluster] = df.Country.unique()\n",
    "    print(f'{cluster} cluster',df.Country.unique())\n",
    "    !mkdir -p output/cluster/$cluster\n",
    "    for country in df.Country.unique():\n",
    "        df2 = clusters[(clusters['Country'] == country) & (clusters['Cluster'] == cluster)]\n",
    "        df2.to_csv(f'output/cluster/{cluster}/{country}.txt', sep='\\t', columns=['SampleID'], header=False, index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "b987e334-48c8-492c-9fe9-4b41c30c46c1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf work/vcf/h3a_wgs.vcf\n",
      "\t--weir-fst-pop output/cluster/Delta/Guinea.txt\n",
      "\t--weir-fst-pop output/cluster/Delta/DRC.txt\n",
      "\t--weir-fst-pop output/cluster/Delta/Cote D'Ivoire.txt\n",
      "\t--keep output/cluster/Delta/Guinea.txt\n",
      "\t--keep output/cluster/Delta/DRC.txt\n",
      "\t--keep output/cluster/Delta/Cote D'Ivoire.txt\n",
      "\t--out output/fst/Delta\n",
      "\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 61 out of 386 Individuals\n",
      "Outputting Weir and Cockerham Fst estimates.\n",
      "Weir and Cockerham mean Fst estimate: 0.0037313\n",
      "Weir and Cockerham weighted Fst estimate: 0.0042855\n",
      "After filtering, kept 1252340 out of a possible 1252340 Sites\n",
      "Run Time = 135.00 seconds\n",
      "CPU times: user 2.16 s, sys: 432 ms, total: 2.59 s\n",
      "Wall time: 2min 15s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "!vcftools --vcf work/vcf/h3a_wgs.vcf --weir-fst-pop output/cluster/Delta/Guinea.txt --weir-fst-pop output/cluster/Delta/DRC.txt --weir-fst-pop output/cluster/Delta/Cote\\ D\\'Ivoire.txt --out output/fst/Delta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "d30ad72e-2b1c-488e-b0f0-e5a45d16e8c3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf work/vcf/h3a_wgs.vcf\n",
      "\t--weir-fst-pop output/cluster/Beta/Uganda.txt\n",
      "\t--weir-fst-pop output/cluster/Beta/Uganda.txt\n",
      "\t--keep output/cluster/Beta/Uganda.txt\n",
      "\t--keep output/cluster/Beta/Uganda.txt\n",
      "\t--out output/fst/Beta\n",
      "\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 51 out of 386 Individuals\n",
      "Outputting Weir and Cockerham Fst estimates.\n",
      "Weir and Cockerham mean Fst estimate: -0.010515\n",
      "Weir and Cockerham weighted Fst estimate: -0.010632\n",
      "After filtering, kept 1252340 out of a possible 1252340 Sites\n",
      "Run Time = 133.00 seconds\n",
      "CPU times: user 2.19 s, sys: 407 ms, total: 2.59 s\n",
      "Wall time: 2min 13s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "!vcftools --vcf work/vcf/h3a_wgs.vcf --weir-fst-pop output/cluster/Beta/Uganda.txt --weir-fst-pop output/cluster/Beta/Uganda.txt --out output/fst/Beta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "62a26b62-cb73-450b-b7b7-c6ca6c7878a9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf work/vcf/h3a_wgs.vcf\n",
      "\t--weir-fst-pop output/cluster/Gamma/Burkina Faso.txt\n",
      "\t--weir-fst-pop output/cluster/Gamma/Nigeria.txt\n",
      "\t--weir-fst-pop output/cluster/Gamma/Cameroon.txt\n",
      "\t--weir-fst-pop output/cluster/Gamma/Ghana.txt\n",
      "\t--weir-fst-pop output/cluster/Gamma/Benin.txt\n",
      "\t--keep output/cluster/Gamma/Burkina Faso.txt\n",
      "\t--keep output/cluster/Gamma/Nigeria.txt\n",
      "\t--keep output/cluster/Gamma/Cameroon.txt\n",
      "\t--keep output/cluster/Gamma/Ghana.txt\n",
      "\t--keep output/cluster/Gamma/Benin.txt\n",
      "\t--out output/fst/Gamma\n",
      "\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 148 out of 386 Individuals\n",
      "Outputting Weir and Cockerham Fst estimates.\n",
      "Weir and Cockerham mean Fst estimate: 0.0033626\n",
      "Weir and Cockerham weighted Fst estimate: 0.0035109\n",
      "After filtering, kept 1252340 out of a possible 1252340 Sites\n",
      "Run Time = 140.00 seconds\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "!vcftools --vcf work/vcf/h3a_wgs.vcf --weir-fst-pop output/cluster/Gamma/Burkina\\ Faso.txt --weir-fst-pop output/cluster/Gamma/Nigeria.txt --weir-fst-pop output/cluster/Gamma/Cameroon.txt --weir-fst-pop output/cluster/Gamma/Ghana.txt --weir-fst-pop output/cluster/Gamma/Benin.txt --out output/fst/Gamma"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "986f2429-16cd-4bcb-8fbf-b1afe19381a7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf work/vcf/h3a_wgs.vcf\n",
      "\t--weir-fst-pop output/cluster/Epsilon/Botswana.txt\n",
      "\t--weir-fst-pop output/cluster/Epsilon/Botswana.txt\n",
      "\t--keep output/cluster/Epsilon/Botswana.txt\n",
      "\t--keep output/cluster/Epsilon/Botswana.txt\n",
      "\t--out output/fst/Epsilon\n",
      "\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 46 out of 386 Individuals\n",
      "Outputting Weir and Cockerham Fst estimates.\n",
      "Weir and Cockerham mean Fst estimate: -0.010819\n",
      "Weir and Cockerham weighted Fst estimate: -0.010791\n",
      "After filtering, kept 1252340 out of a possible 1252340 Sites\n",
      "Run Time = 132.00 seconds\n",
      "CPU times: user 2.05 s, sys: 500 ms, total: 2.55 s\n",
      "Wall time: 2min 12s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "!vcftools --vcf work/vcf/h3a_wgs.vcf --weir-fst-pop output/cluster/Epsilon/Botswana.txt --weir-fst-pop output/cluster/Epsilon/Botswana.txt --out output/fst/Epsilon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "e8376d76-d737-4496-b556-cb0440513100",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--vcf work/vcf/h3a_wgs.vcf\n",
      "\t--weir-fst-pop output/cluster/Alpha/Botswana.txt\n",
      "\t--weir-fst-pop output/cluster/Alpha/Cameroon.txt\n",
      "\t--weir-fst-pop output/cluster/Alpha/Zambia.txt\n",
      "\t--keep output/cluster/Alpha/Botswana.txt\n",
      "\t--keep output/cluster/Alpha/Cameroon.txt\n",
      "\t--keep output/cluster/Alpha/Zambia.txt\n",
      "\t--out output/fst/Alpha\n",
      "\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=PR,Number=0,Type=Flag,Description=\"Provisional reference allele, may not be based on real reference genome\">\n",
      "Keeping individuals in 'keep' list\n",
      "After filtering, kept 80 out of 386 Individuals\n",
      "Outputting Weir and Cockerham Fst estimates.\n",
      "Weir and Cockerham mean Fst estimate: 0.0019482\n",
      "Weir and Cockerham weighted Fst estimate: 0.0024303\n",
      "After filtering, kept 1252340 out of a possible 1252340 Sites\n",
      "Run Time = 135.00 seconds\n",
      "CPU times: user 2.16 s, sys: 442 ms, total: 2.6 s\n",
      "Wall time: 2min 14s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "!vcftools --vcf work/vcf/h3a_wgs.vcf --weir-fst-pop output/cluster/Alpha/Botswana.txt --weir-fst-pop output/cluster/Alpha/Cameroon.txt --weir-fst-pop output/cluster/Alpha/Zambia.txt --out output/fst/Alpha"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "1865c7a0-9f19-49da-8941-6aa73d194b60",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Picked up _JAVA_OPTIONS: -Xmx100g\n",
      "Picked up _JAVA_OPTIONS: -Xmx100g\n",
      "Picked up _JAVA_OPTIONS: -Xmx100g\n",
      "Picked up _JAVA_OPTIONS: -Xmx100g\n",
      "\n",
      "\u001b[32mUsage:\u001b[m\u001b[33m   opencga.sh cohorts create [options]\u001b[m\n",
      "\n",
      "\u001b[32mOptions:\u001b[m\n",
      "                 \u001b[33m--attributes\u001b[m\t\u001b[32mSTRING=OBJECT\u001b[m\t\u001b[32m\u001b[32mThe body web service attributes parameter. Use: --attributes key=value \u001b[m\u001b[m\n",
      "        \u001b[33m--creation-date, --cd\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mThe body web service creationDate parameter \u001b[m\u001b[m\n",
      "                \u001b[33m--description\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mThe body web service description parameter \u001b[m\u001b[m\n",
      "                \u001b[33m--exclude, -E\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mFields excluded in the response, whole JSON path must be provided \u001b[m\u001b[m\n",
      "                   \u001b[33m-h, --help\u001b[m\t\u001b[32m\u001b[m\t\u001b[32m\u001b[32mPrint this help  [false]\u001b[m\u001b[m\n",
      "                       \u001b[33m--host\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mUse this server host instead of the default one in the client configuration \u001b[m\u001b[m\n",
      "                        \u001b[33m*--id\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mThe body web service id parameter \u001b[m\u001b[m\n",
      "                \u001b[33m--include, -I\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mFields included in the response, whole JSON path must be provided \u001b[m\u001b[m\n",
      "             \u001b[33m--include-result\u001b[m\t\u001b[32m\u001b[m\t\u001b[32mFlag indicating to include the created or updated document result in the\u001b[m\n",
      "                                      \u001b[32mresponse  [false]\u001b[m\n",
      "            \u001b[33m--json-data-model\u001b[m\t\u001b[32m\u001b[m\t\u001b[32m\u001b[32mShow example of file structure for body data.  [false]\u001b[m\u001b[m\n",
      "                  \u001b[33m--json-file\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32mFile with the body data in JSON format. Note, that using this parameter will\u001b[m\n",
      "                                      \u001b[32mignore all the other parameters.\u001b[m\n",
      "              \u001b[33m-L, --log-level\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mOne of the following: 'error', 'warn', 'info', 'debug', 'trace' [info] \u001b[m\u001b[m\n",
      "    \u001b[33m--modification-date, --md\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mThe body web service modificationDate parameter \u001b[m\u001b[m\n",
      "                   \u001b[33m--name, -n\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mThe body web service name parameter \u001b[m\u001b[m\n",
      "                  \u001b[33m--no-header\u001b[m\t\u001b[32m\u001b[m\t\u001b[32mNot include headers in the output (not applicable to json output-format)\u001b[m\n",
      "                                      \u001b[32m[false]\u001b[m\n",
      "        \u001b[33m--of, --output-format\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mOutput format. one of {JSON, JSON_PRETTY, TEXT, YAML}  [TEXT]\u001b[m\u001b[m\n",
      "         \u001b[33m--status-description\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mThe body web service description parameter \u001b[m\u001b[m\n",
      "                  \u001b[33m--status-id\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mThe body web service id parameter \u001b[m\u001b[m\n",
      "                \u001b[33m--status-name\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mThe body web service name parameter \u001b[m\u001b[m\n",
      "                  \u001b[33m--study, -s\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32mStudy [[user@]project:]study where study and project can be either the ID or\u001b[m\n",
      "                                      \u001b[32mUUID\u001b[m\n",
      "                       \u001b[33m--type\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32mEnum param allowed values: CASE_CONTROL, CASE_SET, CONTROL_SET, PAIRED,\u001b[m\n",
      "                                      \u001b[32mPAIRED_TUMOR, AGGREGATE, TIME_SERIES, FAMILY, TRIO, COLLECTION\u001b[m\n",
      "                   \u001b[33m--variable\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mDeprecated: Use /generate web service and filter by annotation \u001b[m\u001b[m\n",
      "               \u001b[33m--variable-set\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mDeprecated: Use /generate web service and filter by annotation \u001b[m\u001b[m\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!opencga.sh cohorts  create --help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "a91ca066-eae2-4be7-a7aa-0373f78d0f65",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Picked up _JAVA_OPTIONS: -Xmx100g\n",
      "Picked up _JAVA_OPTIONS: -Xmx100g\n",
      "Picked up _JAVA_OPTIONS: -Xmx100g\n",
      "Picked up _JAVA_OPTIONS: -Xmx100g\n",
      "\u001b[32mINFO: Items processed: 1 created\u001b[m\n"
     ]
    }
   ],
   "source": [
    "!opencga.sh cohorts create --id alpha --name alpha --study wilson@agvp:h3awgs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "f0ed8ca9-c4e5-4856-8e1e-afa91f316e66",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Picked up _JAVA_OPTIONS: -Xmx100g\n",
      "Picked up _JAVA_OPTIONS: -Xmx100g\n",
      "Picked up _JAVA_OPTIONS: -Xmx100g\n",
      "Picked up _JAVA_OPTIONS: -Xmx100g\n",
      "[ {\n",
      "  \"time\" : 0,\n",
      "  \"events\" : [ ],\n",
      "  \"numResults\" : 1,\n",
      "  \"results\" : [ {\n",
      "    \"id\" : \"epsilon\",\n",
      "    \"annotationSets\" : [ ],\n",
      "    \"name\" : \"epsilon\",\n",
      "    \"uuid\" : \"962fb128-0188-0005-0001-a8955966cae3\",\n",
      "    \"type\" : \"COLLECTION\",\n",
      "    \"creationDate\" : \"20230607160807\",\n",
      "    \"modificationDate\" : \"20230607160807\",\n",
      "    \"description\" : \"\",\n",
      "    \"samples\" : [ {\n",
      "      \"id\" : \"H3A-D3A6FE\",\n",
      "      \"uuid\" : \"3bf1abc8-0187-0004-0001-7666ac006345\",\n",
      "      \"source\" : {\n",
      "        \"id\" : \"\",\n",
      "        \"name\" : \"\",\n",
      "        \"description\" : \"\",\n",
      "        \"source\" : \"\",\n",
      "        \"url\" : \"\"\n",
      "      },\n",
      "      \"processing\" : {\n",
      "        \"product\" : {\n",
      "          \"id\" : \"\",\n",
      "          \"name\" : \"\",\n",
      "          \"description\" : \"\",\n",
      "          \"source\" : \"\",\n",
      "          \"url\" : \"\",\n",
      "          \"attributes\" : { }\n",
      "        },\n",
      "        \"preparationMethod\" : \"\",\n",
      "        \"extractionMethod\" : \"\",\n",
      "        \"labSampleId\" : \"\",\n",
      "        \"quantity\" : \"\",\n",
      "        \"date\" : \"\",\n",
      "        \"attributes\" : { }\n",
      "      },\n",
      "      \"collection\" : {\n",
      "        \"from\" : [ ],\n",
      "        \"type\" : \"\",\n",
      "        \"quantity\" : \"\",\n",
      "        \"method\" : \"\",\n",
      "        \"date\" : \"\",\n",
      "        \"attributes\" : { }\n",
      "      },\n",
      "      \"qualityControl\" : {\n",
      "        \"files\" : [ ],\n",
      "        \"comments\" : [ ],\n",
      "        \"variant\" : {\n",
      "          \"variantStats\" : [ ],\n",
      "          \"signatures\" : [ ],\n",
      "          \"hrDetects\" : [ ],\n",
      "          \"files\" : [ ]\n",
      "        }\n",
      "      },\n",
      "      \"release\" : 1,\n",
      "      \"version\" : 11,\n",
      "      \"creationDate\" : \"20230401103146\",\n",
      "      \"modificationDate\" : \"20230518074625\",\n",
      "      \"description\" : \"\",\n",
      "      \"somatic\" : false,\n",
      "      \"phenotypes\" : [ ],\n",
      "      \"individualId\" : \"\",\n",
      "      \"fileIds\" : [ \"h3a_wgs.gtypes.vcf.gz\" ],\n",
      "      \"cohortIds\" : [ \"ALL\", \"alpha\", \"epsilon\" ],\n",
      "      \"status\" : {\n",
      "        \"id\" : \"\",\n",
      "        \"name\" : \"\",\n",
      "        \"description\" : \"\",\n",
      "        \"date\" : \"\"\n",
      "      },\n",
      "      \"internal\" : {\n",
      "        \"status\" : {\n",
      "          \"id\" : \"READY\",\n",
      "          \"name\" : \"READY\",\n",
      "          \"description\" : \"\",\n",
      "          \"date\" : \"20230401103146\",\n",
      "          \"version\" : \"2.7.1-SNAPSHOT\",\n",
      "          \"commit\" : \"8d7356b4f58cbc4564cdd931aa59154aff4c6f42\"\n",
      "        },\n",
      "        \"registrationDate\" : \"20230401103146\",\n",
      "        \"lastModified\" : \"20230518074625\",\n",
      "        \"rga\" : {\n",
      "          \"status\" : \"NOT_INDEXED\",\n",
      "          \"date\" : \"20230401103146\"\n",
      "        },\n",
      "        \"variant\" : {\n",
      "          \"index\" : {\n",
      "            \"status\" : {\n",
      "              \"id\" : \"READY\",\n",
      "              \"name\" : \"READY\",\n",
      "              \"description\" : \"\",\n",
      "              \"date\" : \"20230403194450\",\n",
      "              \"version\" : \"2.7.1-SNAPSHOT\",\n",
      "              \"commit\" : \"8d7356b4f58cbc4564cdd931aa59154aff4c6f42\"\n",
      "            },\n",
      "            \"numFiles\" : 1,\n",
      "            \"multiFile\" : false\n",
      "          },\n",
      "          \"secondarySampleIndex\" : {\n",
      "            \"status\" : {\n",
      "              \"id\" : \"NONE\",\n",
      "              \"name\" : \"NONE\",\n",
      "              \"description\" : \"SecondarySampleIndex is not ready with version=1, genotypeStatus=NONE, annotationStatus=READY\",\n",
      "              \"date\" : \"20230518074625\",\n",
      "              \"version\" : \"2.6.2-SNAPSHOT\",\n",
      "              \"commit\" : \"2d0ac9b7254947af9179ac6428d2e611590ad2a1\"\n",
      "            },\n",
      "            \"familyStatus\" : {\n",
      "              \"id\" : \"NONE\",\n",
      "              \"name\" : \"NONE\",\n",
      "              \"description\" : \"Family Index is not ready with version=1\",\n",
      "              \"date\" : \"20230403165512\",\n",
      "              \"version\" : \"2.7.1-SNAPSHOT\",\n",
      "              \"commit\" : \"8d7356b4f58cbc4564cdd931aa59154aff4c6f42\"\n",
      "            }\n",
      "          },\n",
      "          \"annotationIndex\" : {\n",
      "            \"status\" : {\n",
      "              \"id\" : \"READY\",\n",
      "              \"name\" : \"READY\",\n",
      "              \"description\" : \"\",\n",
      "              \"date\" : \"20230518074625\",\n",
      "              \"version\" : \"2.6.2-SNAPSHOT\",\n",
      "              \"commit\" : \"2d0ac9b7254947af9179ac6428d2e611590ad2a1\"\n",
      "            }\n",
      "          },\n",
      "          \"secondaryAnnotationIndex\" : {\n",
      "            \"status\" : {\n",
      "              \"id\" : \"NONE\",\n",
      "              \"name\" : \"NONE\",\n",
      "              \"description\" : \"\",\n",
      "              \"date\" : \"20230403165512\",\n",
      "              \"version\" : \"2.7.1-SNAPSHOT\",\n",
      "              \"commit\" : \"8d7356b4f58cbc4564cdd931aa59154aff4c6f42\"\n",
      "            }\n",
      "          }\n",
      "        }\n",
      "      },\n",
      "      \"attributes\" : { }\n",
      "    } ],\n",
      "    \"numSamples\" : 1,\n",
      "    \"release\" : 1,\n",
      "    \"status\" : {\n",
      "      \"id\" : \"\",\n",
      "      \"name\" : \"\",\n",
      "      \"description\" : \"\",\n",
      "      \"date\" : \"\"\n",
      "    },\n",
      "    \"internal\" : {\n",
      "      \"status\" : {\n",
      "        \"id\" : \"NONE\",\n",
      "        \"name\" : \"NONE\",\n",
      "        \"description\" : \"\",\n",
      "        \"date\" : \"20230607160807\",\n",
      "        \"version\" : \"2.6.2-SNAPSHOT\",\n",
      "        \"commit\" : \"2d0ac9b7254947af9179ac6428d2e611590ad2a1\"\n",
      "      },\n",
      "      \"registrationDate\" : \"20230607160807\",\n",
      "      \"lastModified\" : \"20230607160807\"\n",
      "    },\n",
      "    \"attributes\" : { }\n",
      "  } ],\n",
      "  \"resultType\" : \"\",\n",
      "  \"numMatches\" : 1,\n",
      "  \"numInserted\" : 0,\n",
      "  \"numUpdated\" : 0,\n",
      "  \"numDeleted\" : 0,\n",
      "  \"numErrors\" : 0,\n",
      "  \"attributes\" : { },\n",
      "  \"numTotalResults\" : 1,\n",
      "  \"federationNode\" : {\n",
      "    \"id\" : \"primary\",\n",
      "    \"uri\" : \"http://localhost:8080/opencga-2.6.2-SNAPSHOT/webservices/rest/\",\n",
      "    \"commit\" : \"2d0ac9b7254947af9179ac6428d2e611590ad2a1\",\n",
      "    \"version\" : \"2.6.2-SNAPSHOT\"\n",
      "  }\n",
      "} ]\n"
     ]
    }
   ],
   "source": [
    "!opencga.sh cohorts info --cohorts epsilon --study wilson@agvp:h3awgs --output-format JSON_PRETTY"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "4cc1e323-44ab-43ec-883a-25272001a0d7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Picked up _JAVA_OPTIONS: -Xmx100g\n",
      "Picked up _JAVA_OPTIONS: -Xmx100g\n",
      "Picked up _JAVA_OPTIONS: -Xmx100g\n",
      "Picked up _JAVA_OPTIONS: -Xmx100g\n",
      "\n",
      "\u001b[32mUsage:\u001b[m\u001b[33m   opencga.sh cohorts update [options]\u001b[m\n",
      "\n",
      "\u001b[32mOptions:\u001b[m\n",
      "                 \u001b[33m--attributes\u001b[m\t\u001b[32mSTRING=OBJECT\u001b[m\t\u001b[32m\u001b[32mThe body web service attributes parameter. Use: --attributes key=value \u001b[m\u001b[m\n",
      "                   \u001b[33m*--cohorts\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mComma separated list of cohort ids \u001b[m\u001b[m\n",
      "        \u001b[33m--creation-date, --cd\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mThe body web service creationDate parameter \u001b[m\u001b[m\n",
      "                \u001b[33m--description\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mThe body web service description parameter \u001b[m\u001b[m\n",
      "                \u001b[33m--exclude, -E\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mFields excluded in the response, whole JSON path must be provided \u001b[m\u001b[m\n",
      "                   \u001b[33m-h, --help\u001b[m\t\u001b[32m\u001b[m\t\u001b[32m\u001b[32mPrint this help  [false]\u001b[m\u001b[m\n",
      "                       \u001b[33m--host\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mUse this server host instead of the default one in the client configuration \u001b[m\u001b[m\n",
      "                         \u001b[33m--id\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mThe body web service id parameter \u001b[m\u001b[m\n",
      "                \u001b[33m--include, -I\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mFields included in the response, whole JSON path must be provided \u001b[m\u001b[m\n",
      "             \u001b[33m--include-result\u001b[m\t\u001b[32m\u001b[m\t\u001b[32mFlag indicating to include the created or updated document result in the\u001b[m\n",
      "                                      \u001b[32mresponse  [false]\u001b[m\n",
      "            \u001b[33m--json-data-model\u001b[m\t\u001b[32m\u001b[m\t\u001b[32m\u001b[32mShow example of file structure for body data.  [false]\u001b[m\u001b[m\n",
      "                  \u001b[33m--json-file\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32mFile with the body data in JSON format. Note, that using this parameter will\u001b[m\n",
      "                                      \u001b[32mignore all the other parameters.\u001b[m\n",
      "              \u001b[33m-L, --log-level\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mOne of the following: 'error', 'warn', 'info', 'debug', 'trace' [info] \u001b[m\u001b[m\n",
      "    \u001b[33m--modification-date, --md\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mThe body web service modificationDate parameter \u001b[m\u001b[m\n",
      "                   \u001b[33m--name, -n\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mThe body web service name parameter \u001b[m\u001b[m\n",
      "                  \u001b[33m--no-header\u001b[m\t\u001b[32m\u001b[m\t\u001b[32mNot include headers in the output (not applicable to json output-format)\u001b[m\n",
      "                                      \u001b[32m[false]\u001b[m\n",
      "        \u001b[33m--of, --output-format\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mOutput format. one of {JSON, JSON_PRETTY, TEXT, YAML}  [TEXT]\u001b[m\u001b[m\n",
      "         \u001b[33m--status-description\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mThe body web service description parameter \u001b[m\u001b[m\n",
      "                  \u001b[33m--status-id\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mThe body web service id parameter \u001b[m\u001b[m\n",
      "                \u001b[33m--status-name\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mThe body web service name parameter \u001b[m\u001b[m\n",
      "                  \u001b[33m--study, -s\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32mStudy [[user@]project:]study where study and project can be either the ID or\u001b[m\n",
      "                                      \u001b[32mUUID\u001b[m\n",
      "                       \u001b[33m--type\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32mEnum param allowed values: CASE_CONTROL, CASE_SET, CONTROL_SET, PAIRED,\u001b[m\n",
      "                                      \u001b[32mPAIRED_TUMOR, AGGREGATE, TIME_SERIES, FAMILY, TRIO, COLLECTION\u001b[m\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!opencga.sh cohorts  update --help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "6d35104a-2e65-46b3-82d4-9cc8dab9d573",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Picked up _JAVA_OPTIONS: -Xmx100g\n",
      "Picked up _JAVA_OPTIONS: -Xmx100g\n",
      "Picked up _JAVA_OPTIONS: -Xmx100g\n",
      "Picked up _JAVA_OPTIONS: -Xmx100g\n",
      "\u001b[31mERROR: Was passed main parameter '--samples' but no main parameter was defined in your arg class\u001b[m\n",
      "\n",
      "\u001b[32mUsage:\u001b[m\u001b[33m   opencga.sh cohorts create [options]\u001b[m\n",
      "\n",
      "\u001b[32mOptions:\u001b[m\n",
      "                 \u001b[33m--attributes\u001b[m\t\u001b[32mSTRING=OBJECT\u001b[m\t\u001b[32m\u001b[32mThe body web service attributes parameter. Use: --attributes key=value \u001b[m\u001b[m\n",
      "        \u001b[33m--creation-date, --cd\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mThe body web service creationDate parameter \u001b[m\u001b[m\n",
      "                \u001b[33m--description\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mThe body web service description parameter \u001b[m\u001b[m\n",
      "                \u001b[33m--exclude, -E\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mFields excluded in the response, whole JSON path must be provided \u001b[m\u001b[m\n",
      "                   \u001b[33m-h, --help\u001b[m\t\u001b[32m\u001b[m\t\u001b[32m\u001b[32mPrint this help  [false]\u001b[m\u001b[m\n",
      "                       \u001b[33m--host\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mUse this server host instead of the default one in the client configuration \u001b[m\u001b[m\n",
      "                        \u001b[33m*--id\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mThe body web service id parameter \u001b[m\u001b[m\n",
      "                \u001b[33m--include, -I\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mFields included in the response, whole JSON path must be provided \u001b[m\u001b[m\n",
      "             \u001b[33m--include-result\u001b[m\t\u001b[32m\u001b[m\t\u001b[32mFlag indicating to include the created or updated document result in the\u001b[m\n",
      "                                      \u001b[32mresponse  [false]\u001b[m\n",
      "            \u001b[33m--json-data-model\u001b[m\t\u001b[32m\u001b[m\t\u001b[32m\u001b[32mShow example of file structure for body data.  [false]\u001b[m\u001b[m\n",
      "                  \u001b[33m--json-file\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32mFile with the body data in JSON format. Note, that using this parameter will\u001b[m\n",
      "                                      \u001b[32mignore all the other parameters.\u001b[m\n",
      "              \u001b[33m-L, --log-level\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mOne of the following: 'error', 'warn', 'info', 'debug', 'trace' [info] \u001b[m\u001b[m\n",
      "    \u001b[33m--modification-date, --md\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mThe body web service modificationDate parameter \u001b[m\u001b[m\n",
      "                   \u001b[33m--name, -n\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mThe body web service name parameter \u001b[m\u001b[m\n",
      "                  \u001b[33m--no-header\u001b[m\t\u001b[32m\u001b[m\t\u001b[32mNot include headers in the output (not applicable to json output-format)\u001b[m\n",
      "                                      \u001b[32m[false]\u001b[m\n",
      "        \u001b[33m--of, --output-format\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mOutput format. one of {JSON, JSON_PRETTY, TEXT, YAML}  [TEXT]\u001b[m\u001b[m\n",
      "         \u001b[33m--status-description\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mThe body web service description parameter \u001b[m\u001b[m\n",
      "                  \u001b[33m--status-id\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mThe body web service id parameter \u001b[m\u001b[m\n",
      "                \u001b[33m--status-name\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mThe body web service name parameter \u001b[m\u001b[m\n",
      "                  \u001b[33m--study, -s\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32mStudy [[user@]project:]study where study and project can be either the ID or\u001b[m\n",
      "                                      \u001b[32mUUID\u001b[m\n",
      "                       \u001b[33m--type\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32mEnum param allowed values: CASE_CONTROL, CASE_SET, CONTROL_SET, PAIRED,\u001b[m\n",
      "                                      \u001b[32mPAIRED_TUMOR, AGGREGATE, TIME_SERIES, FAMILY, TRIO, COLLECTION\u001b[m\n",
      "                   \u001b[33m--variable\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mDeprecated: Use /generate web service and filter by annotation \u001b[m\u001b[m\n",
      "               \u001b[33m--variable-set\u001b[m\t\u001b[32mSTRING\u001b[m\t\u001b[32m\u001b[32mDeprecated: Use /generate web service and filter by annotation \u001b[m\u001b[m\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!opencga.sh cohorts create --name beta --study wilson@agvp:h3awgs --samples H3A-31F23C,H3A-D3A6FE,H3A-8CE927"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "429b1e8d-c545-454d-87ac-45ab6cc6a954",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Picked up _JAVA_OPTIONS: -Xmx100g\n",
      "Picked up _JAVA_OPTIONS: -Xmx100g\n",
      "Picked up _JAVA_OPTIONS: -Xmx100g\n",
      "Picked up _JAVA_OPTIONS: -Xmx100g\n",
      "[ {\n",
      "  \"time\" : 0,\n",
      "  \"events\" : [ ],\n",
      "  \"numResults\" : 3,\n",
      "  \"results\" : [ {\n",
      "    \"id\" : \"H3A-31F23C\",\n",
      "    \"annotationSets\" : [ ],\n",
      "    \"uuid\" : \"3bf1ab95-0187-0004-0001-b8929200aaf0\",\n",
      "    \"source\" : {\n",
      "      \"id\" : \"\",\n",
      "      \"name\" : \"\",\n",
      "      \"description\" : \"\",\n",
      "      \"source\" : \"\",\n",
      "      \"url\" : \"\"\n",
      "    },\n",
      "    \"processing\" : {\n",
      "      \"product\" : {\n",
      "        \"id\" : \"\",\n",
      "        \"name\" : \"\",\n",
      "        \"description\" : \"\",\n",
      "        \"source\" : \"\",\n",
      "        \"url\" : \"\",\n",
      "        \"attributes\" : { }\n",
      "      },\n",
      "      \"preparationMethod\" : \"\",\n",
      "      \"extractionMethod\" : \"\",\n",
      "      \"labSampleId\" : \"\",\n",
      "      \"quantity\" : \"\",\n",
      "      \"date\" : \"\",\n",
      "      \"attributes\" : { }\n",
      "    },\n",
      "    \"collection\" : {\n",
      "      \"from\" : [ ],\n",
      "      \"type\" : \"\",\n",
      "      \"quantity\" : \"\",\n",
      "      \"method\" : \"\",\n",
      "      \"date\" : \"\",\n",
      "      \"attributes\" : { }\n",
      "    },\n",
      "    \"qualityControl\" : {\n",
      "      \"files\" : [ ],\n",
      "      \"comments\" : [ ],\n",
      "      \"variant\" : {\n",
      "        \"variantStats\" : [ ],\n",
      "        \"signatures\" : [ ],\n",
      "        \"hrDetects\" : [ ],\n",
      "        \"files\" : [ ]\n",
      "      }\n",
      "    },\n",
      "    \"release\" : 1,\n",
      "    \"version\" : 9,\n",
      "    \"creationDate\" : \"20230401103146\",\n",
      "    \"modificationDate\" : \"20230518074605\",\n",
      "    \"description\" : \"\",\n",
      "    \"somatic\" : false,\n",
      "    \"phenotypes\" : [ ],\n",
      "    \"individualId\" : \"\",\n",
      "    \"fileIds\" : [ \"h3a_wgs.gtypes.vcf.gz\" ],\n",
      "    \"cohortIds\" : [ \"ALL\" ],\n",
      "    \"status\" : {\n",
      "      \"id\" : \"\",\n",
      "      \"name\" : \"\",\n",
      "      \"description\" : \"\",\n",
      "      \"date\" : \"\"\n",
      "    },\n",
      "    \"internal\" : {\n",
      "      \"status\" : {\n",
      "        \"id\" : \"READY\",\n",
      "        \"name\" : \"READY\",\n",
      "        \"description\" : \"\",\n",
      "        \"date\" : \"20230401103146\",\n",
      "        \"version\" : \"2.7.1-SNAPSHOT\",\n",
      "        \"commit\" : \"8d7356b4f58cbc4564cdd931aa59154aff4c6f42\"\n",
      "      },\n",
      "      \"registrationDate\" : \"20230401103146\",\n",
      "      \"lastModified\" : \"20230518074605\",\n",
      "      \"rga\" : {\n",
      "        \"status\" : \"NOT_INDEXED\",\n",
      "        \"date\" : \"20230401103146\"\n",
      "      },\n",
      "      \"variant\" : {\n",
      "        \"index\" : {\n",
      "          \"status\" : {\n",
      "            \"id\" : \"READY\",\n",
      "            \"name\" : \"READY\",\n",
      "            \"description\" : \"\",\n",
      "            \"date\" : \"20230403194440\",\n",
      "            \"version\" : \"2.7.1-SNAPSHOT\",\n",
      "            \"commit\" : \"8d7356b4f58cbc4564cdd931aa59154aff4c6f42\"\n",
      "          },\n",
      "          \"numFiles\" : 1,\n",
      "          \"multiFile\" : false\n",
      "        },\n",
      "        \"secondarySampleIndex\" : {\n",
      "          \"status\" : {\n",
      "            \"id\" : \"NONE\",\n",
      "            \"name\" : \"NONE\",\n",
      "            \"description\" : \"SecondarySampleIndex is not ready with version=1, genotypeStatus=NONE, annotationStatus=READY\",\n",
      "            \"date\" : \"20230518074605\",\n",
      "            \"version\" : \"2.6.2-SNAPSHOT\",\n",
      "            \"commit\" : \"2d0ac9b7254947af9179ac6428d2e611590ad2a1\"\n",
      "          },\n",
      "          \"familyStatus\" : {\n",
      "            \"id\" : \"NONE\",\n",
      "            \"name\" : \"NONE\",\n",
      "            \"description\" : \"Family Index is not ready with version=1\",\n",
      "            \"date\" : \"20230403165436\",\n",
      "            \"version\" : \"2.7.1-SNAPSHOT\",\n",
      "            \"commit\" : \"8d7356b4f58cbc4564cdd931aa59154aff4c6f42\"\n",
      "          }\n",
      "        },\n",
      "        \"annotationIndex\" : {\n",
      "          \"status\" : {\n",
      "            \"id\" : \"READY\",\n",
      "            \"name\" : \"READY\",\n",
      "            \"description\" : \"\",\n",
      "            \"date\" : \"20230518074605\",\n",
      "            \"version\" : \"2.6.2-SNAPSHOT\",\n",
      "            \"commit\" : \"2d0ac9b7254947af9179ac6428d2e611590ad2a1\"\n",
      "          }\n",
      "        },\n",
      "        \"secondaryAnnotationIndex\" : {\n",
      "          \"status\" : {\n",
      "            \"id\" : \"NONE\",\n",
      "            \"name\" : \"NONE\",\n",
      "            \"description\" : \"\",\n",
      "            \"date\" : \"20230403165436\",\n",
      "            \"version\" : \"2.7.1-SNAPSHOT\",\n",
      "            \"commit\" : \"8d7356b4f58cbc4564cdd931aa59154aff4c6f42\"\n",
      "          }\n",
      "        }\n",
      "      }\n",
      "    },\n",
      "    \"attributes\" : { }\n",
      "  }, {\n",
      "    \"id\" : \"H3A-D3A6FE\",\n",
      "    \"annotationSets\" : [ ],\n",
      "    \"uuid\" : \"3bf1abc8-0187-0004-0001-7666ac006345\",\n",
      "    \"source\" : {\n",
      "      \"id\" : \"\",\n",
      "      \"name\" : \"\",\n",
      "      \"description\" : \"\",\n",
      "      \"source\" : \"\",\n",
      "      \"url\" : \"\"\n",
      "    },\n",
      "    \"processing\" : {\n",
      "      \"product\" : {\n",
      "        \"id\" : \"\",\n",
      "        \"name\" : \"\",\n",
      "        \"description\" : \"\",\n",
      "        \"source\" : \"\",\n",
      "        \"url\" : \"\",\n",
      "        \"attributes\" : { }\n",
      "      },\n",
      "      \"preparationMethod\" : \"\",\n",
      "      \"extractionMethod\" : \"\",\n",
      "      \"labSampleId\" : \"\",\n",
      "      \"quantity\" : \"\",\n",
      "      \"date\" : \"\",\n",
      "      \"attributes\" : { }\n",
      "    },\n",
      "    \"collection\" : {\n",
      "      \"from\" : [ ],\n",
      "      \"type\" : \"\",\n",
      "      \"quantity\" : \"\",\n",
      "      \"method\" : \"\",\n",
      "      \"date\" : \"\",\n",
      "      \"attributes\" : { }\n",
      "    },\n",
      "    \"qualityControl\" : {\n",
      "      \"files\" : [ ],\n",
      "      \"comments\" : [ ],\n",
      "      \"variant\" : {\n",
      "        \"variantStats\" : [ ],\n",
      "        \"signatures\" : [ ],\n",
      "        \"hrDetects\" : [ ],\n",
      "        \"files\" : [ ]\n",
      "      }\n",
      "    },\n",
      "    \"release\" : 1,\n",
      "    \"version\" : 9,\n",
      "    \"creationDate\" : \"20230401103146\",\n",
      "    \"modificationDate\" : \"20230518074625\",\n",
      "    \"description\" : \"\",\n",
      "    \"somatic\" : false,\n",
      "    \"phenotypes\" : [ ],\n",
      "    \"individualId\" : \"\",\n",
      "    \"fileIds\" : [ \"h3a_wgs.gtypes.vcf.gz\" ],\n",
      "    \"cohortIds\" : [ \"ALL\" ],\n",
      "    \"status\" : {\n",
      "      \"id\" : \"\",\n",
      "      \"name\" : \"\",\n",
      "      \"description\" : \"\",\n",
      "      \"date\" : \"\"\n",
      "    },\n",
      "    \"internal\" : {\n",
      "      \"status\" : {\n",
      "        \"id\" : \"READY\",\n",
      "        \"name\" : \"READY\",\n",
      "        \"description\" : \"\",\n",
      "        \"date\" : \"20230401103146\",\n",
      "        \"version\" : \"2.7.1-SNAPSHOT\",\n",
      "        \"commit\" : \"8d7356b4f58cbc4564cdd931aa59154aff4c6f42\"\n",
      "      },\n",
      "      \"registrationDate\" : \"20230401103146\",\n",
      "      \"lastModified\" : \"20230518074625\",\n",
      "      \"rga\" : {\n",
      "        \"status\" : \"NOT_INDEXED\",\n",
      "        \"date\" : \"20230401103146\"\n",
      "      },\n",
      "      \"variant\" : {\n",
      "        \"index\" : {\n",
      "          \"status\" : {\n",
      "            \"id\" : \"READY\",\n",
      "            \"name\" : \"READY\",\n",
      "            \"description\" : \"\",\n",
      "            \"date\" : \"20230403194450\",\n",
      "            \"version\" : \"2.7.1-SNAPSHOT\",\n",
      "            \"commit\" : \"8d7356b4f58cbc4564cdd931aa59154aff4c6f42\"\n",
      "          },\n",
      "          \"numFiles\" : 1,\n",
      "          \"multiFile\" : false\n",
      "        },\n",
      "        \"secondarySampleIndex\" : {\n",
      "          \"status\" : {\n",
      "            \"id\" : \"NONE\",\n",
      "            \"name\" : \"NONE\",\n",
      "            \"description\" : \"SecondarySampleIndex is not ready with version=1, genotypeStatus=NONE, annotationStatus=READY\",\n",
      "            \"date\" : \"20230518074625\",\n",
      "            \"version\" : \"2.6.2-SNAPSHOT\",\n",
      "            \"commit\" : \"2d0ac9b7254947af9179ac6428d2e611590ad2a1\"\n",
      "          },\n",
      "          \"familyStatus\" : {\n",
      "            \"id\" : \"NONE\",\n",
      "            \"name\" : \"NONE\",\n",
      "            \"description\" : \"Family Index is not ready with version=1\",\n",
      "            \"date\" : \"20230403165512\",\n",
      "            \"version\" : \"2.7.1-SNAPSHOT\",\n",
      "            \"commit\" : \"8d7356b4f58cbc4564cdd931aa59154aff4c6f42\"\n",
      "          }\n",
      "        },\n",
      "        \"annotationIndex\" : {\n",
      "          \"status\" : {\n",
      "            \"id\" : \"READY\",\n",
      "            \"name\" : \"READY\",\n",
      "            \"description\" : \"\",\n",
      "            \"date\" : \"20230518074625\",\n",
      "            \"version\" : \"2.6.2-SNAPSHOT\",\n",
      "            \"commit\" : \"2d0ac9b7254947af9179ac6428d2e611590ad2a1\"\n",
      "          }\n",
      "        },\n",
      "        \"secondaryAnnotationIndex\" : {\n",
      "          \"status\" : {\n",
      "            \"id\" : \"NONE\",\n",
      "            \"name\" : \"NONE\",\n",
      "            \"description\" : \"\",\n",
      "            \"date\" : \"20230403165512\",\n",
      "            \"version\" : \"2.7.1-SNAPSHOT\",\n",
      "            \"commit\" : \"8d7356b4f58cbc4564cdd931aa59154aff4c6f42\"\n",
      "          }\n",
      "        }\n",
      "      }\n",
      "    },\n",
      "    \"attributes\" : { }\n",
      "  }, {\n",
      "    \"id\" : \"H3A-8CE927\",\n",
      "    \"annotationSets\" : [ ],\n",
      "    \"uuid\" : \"3bf1ac22-0187-0004-0001-514086662992\",\n",
      "    \"source\" : {\n",
      "      \"id\" : \"\",\n",
      "      \"name\" : \"\",\n",
      "      \"description\" : \"\",\n",
      "      \"source\" : \"\",\n",
      "      \"url\" : \"\"\n",
      "    },\n",
      "    \"processing\" : {\n",
      "      \"product\" : {\n",
      "        \"id\" : \"\",\n",
      "        \"name\" : \"\",\n",
      "        \"description\" : \"\",\n",
      "        \"source\" : \"\",\n",
      "        \"url\" : \"\",\n",
      "        \"attributes\" : { }\n",
      "      },\n",
      "      \"preparationMethod\" : \"\",\n",
      "      \"extractionMethod\" : \"\",\n",
      "      \"labSampleId\" : \"\",\n",
      "      \"quantity\" : \"\",\n",
      "      \"date\" : \"\",\n",
      "      \"attributes\" : { }\n",
      "    },\n",
      "    \"collection\" : {\n",
      "      \"from\" : [ ],\n",
      "      \"type\" : \"\",\n",
      "      \"quantity\" : \"\",\n",
      "      \"method\" : \"\",\n",
      "      \"date\" : \"\",\n",
      "      \"attributes\" : { }\n",
      "    },\n",
      "    \"qualityControl\" : {\n",
      "      \"files\" : [ ],\n",
      "      \"comments\" : [ ],\n",
      "      \"variant\" : {\n",
      "        \"variantStats\" : [ ],\n",
      "        \"signatures\" : [ ],\n",
      "        \"hrDetects\" : [ ],\n",
      "        \"files\" : [ ]\n",
      "      }\n",
      "    },\n",
      "    \"release\" : 1,\n",
      "    \"version\" : 9,\n",
      "    \"creationDate\" : \"20230401103146\",\n",
      "    \"modificationDate\" : \"20230518074617\",\n",
      "    \"description\" : \"\",\n",
      "    \"somatic\" : false,\n",
      "    \"phenotypes\" : [ ],\n",
      "    \"individualId\" : \"\",\n",
      "    \"fileIds\" : [ \"h3a_wgs.gtypes.vcf.gz\" ],\n",
      "    \"cohortIds\" : [ \"ALL\" ],\n",
      "    \"status\" : {\n",
      "      \"id\" : \"\",\n",
      "      \"name\" : \"\",\n",
      "      \"description\" : \"\",\n",
      "      \"date\" : \"\"\n",
      "    },\n",
      "    \"internal\" : {\n",
      "      \"status\" : {\n",
      "        \"id\" : \"READY\",\n",
      "        \"name\" : \"READY\",\n",
      "        \"description\" : \"\",\n",
      "        \"date\" : \"20230401103146\",\n",
      "        \"version\" : \"2.7.1-SNAPSHOT\",\n",
      "        \"commit\" : \"8d7356b4f58cbc4564cdd931aa59154aff4c6f42\"\n",
      "      },\n",
      "      \"registrationDate\" : \"20230401103146\",\n",
      "      \"lastModified\" : \"20230518074617\",\n",
      "      \"rga\" : {\n",
      "        \"status\" : \"NOT_INDEXED\",\n",
      "        \"date\" : \"20230401103146\"\n",
      "      },\n",
      "      \"variant\" : {\n",
      "        \"index\" : {\n",
      "          \"status\" : {\n",
      "            \"id\" : \"READY\",\n",
      "            \"name\" : \"READY\",\n",
      "            \"description\" : \"\",\n",
      "            \"date\" : \"20230403194446\",\n",
      "            \"version\" : \"2.7.1-SNAPSHOT\",\n",
      "            \"commit\" : \"8d7356b4f58cbc4564cdd931aa59154aff4c6f42\"\n",
      "          },\n",
      "          \"numFiles\" : 1,\n",
      "          \"multiFile\" : false\n",
      "        },\n",
      "        \"secondarySampleIndex\" : {\n",
      "          \"status\" : {\n",
      "            \"id\" : \"NONE\",\n",
      "            \"name\" : \"NONE\",\n",
      "            \"description\" : \"SecondarySampleIndex is not ready with version=1, genotypeStatus=NONE, annotationStatus=READY\",\n",
      "            \"date\" : \"20230518074617\",\n",
      "            \"version\" : \"2.6.2-SNAPSHOT\",\n",
      "            \"commit\" : \"2d0ac9b7254947af9179ac6428d2e611590ad2a1\"\n",
      "          },\n",
      "          \"familyStatus\" : {\n",
      "            \"id\" : \"NONE\",\n",
      "            \"name\" : \"NONE\",\n",
      "            \"description\" : \"Family Index is not ready with version=1\",\n",
      "            \"date\" : \"20230403165457\",\n",
      "            \"version\" : \"2.7.1-SNAPSHOT\",\n",
      "            \"commit\" : \"8d7356b4f58cbc4564cdd931aa59154aff4c6f42\"\n",
      "          }\n",
      "        },\n",
      "        \"annotationIndex\" : {\n",
      "          \"status\" : {\n",
      "            \"id\" : \"READY\",\n",
      "            \"name\" : \"READY\",\n",
      "            \"description\" : \"\",\n",
      "            \"date\" : \"20230518074617\",\n",
      "            \"version\" : \"2.6.2-SNAPSHOT\",\n",
      "            \"commit\" : \"2d0ac9b7254947af9179ac6428d2e611590ad2a1\"\n",
      "          }\n",
      "        },\n",
      "        \"secondaryAnnotationIndex\" : {\n",
      "          \"status\" : {\n",
      "            \"id\" : \"NONE\",\n",
      "            \"name\" : \"NONE\",\n",
      "            \"description\" : \"\",\n",
      "            \"date\" : \"20230403165457\",\n",
      "            \"version\" : \"2.7.1-SNAPSHOT\",\n",
      "            \"commit\" : \"8d7356b4f58cbc4564cdd931aa59154aff4c6f42\"\n",
      "          }\n",
      "        }\n",
      "      }\n",
      "    },\n",
      "    \"attributes\" : { }\n",
      "  } ],\n",
      "  \"resultType\" : \"\",\n",
      "  \"numMatches\" : 3,\n",
      "  \"numInserted\" : 0,\n",
      "  \"numUpdated\" : 0,\n",
      "  \"numDeleted\" : 0,\n",
      "  \"numErrors\" : 0,\n",
      "  \"attributes\" : { },\n",
      "  \"numTotalResults\" : 3,\n",
      "  \"federationNode\" : {\n",
      "    \"id\" : \"primary\",\n",
      "    \"uri\" : \"http://localhost:8080/opencga-2.6.2-SNAPSHOT/webservices/rest/\",\n",
      "    \"commit\" : \"2d0ac9b7254947af9179ac6428d2e611590ad2a1\",\n",
      "    \"version\" : \"2.6.2-SNAPSHOT\"\n",
      "  }\n",
      "} ]\n"
     ]
    }
   ],
   "source": [
    "!opencga.sh samples info --samples H3A-31F23C,H3A-D3A6FE,H3A-8CE927 -s wilson@agvp:h3awgs --of JSON_PRETTY"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b0b4266e-da39-470f-b1df-9f1fc259cceb",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
